packages/core/src/components/Button/Button.stories.tsx

```diff
@@ -194,3 +194,33 @@ export const Semantic = ({ onClick }) => {
 Semantic.args = {
   onClick: clickAction,
 };
+
+export const SetComponent = ({ onClick }) => {
+  return (
+    <HvBox sx={{ display: "flex", gap: 20, padding: 20 }}>
+      <HvButton onClick={onClick}>Button element</HvButton>
+      <HvButton onClick={onClick} component="a">
+        Link element
+      </HvButton>
+      <HvButton onClick={onClick} component="div">
+        Div element
+      </HvButton>
```

Comment:
I believe that the most common use-cases will be to use the router's `Link`, or a custom button element. Can we add a sample of that?

We can add a dummy one, like so:
```tsx
const Link = ({to, children}) => <a href={to}>{children}</a>
```

Reply:
I will add this